sycl/doc/extensions/Reduction/Reduction_status.md

```diff
@@ -0,0 +1,189 @@
+
+There are 2 specifications of the reduction feature and both are still actual:
+
+* `sycl::ext::oneapi::reduction` is described in [this document](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/Reduction/Reduction.md). This extension was created as part of a pathfinding/prototyping work before it was added to SYCL 2020 standard.
+
+* `sycl::reduction` is described in [SYCL 2020 standard](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:reduction).
+
+These two specifications for reduction are pretty similar. The implementation of `sycl::reduction` is based on (basically re-uses) the implementation of `sycl::ext::oneapi::reduction`. 
+
+There are non-critical differences in API to create the reduction object. `sycl::reduction` accepts either `sycl::buffer` or `usm memory` and optional property `property::reduction::initialize_to_identity` as parameter to create a reduction, while `sycl::ext::oneapi::reduction` accepts `sycl::accessor` that has `access::mode` equal to either `read_write` (which corresponds to SYCL 2020 reduction initialized without `property::reduction::initialize_to_identity`) or `discard_write`(corresponds to case when `property::reduction::initialize_to_identity` is used).
+
+---
+---
+# Implementation details: `reduction` in `parallel_for()` accepting `nd_range`
+
+## Reduction inside 1 work-group - the main building block for reduction
+
+The reduction algorithm for 1 basic block depends currently on combination/availability of 2 features:
+- `fast atomics` (i.e. operations available on the target device), such as fetch_add(), fetch_min(), etc.
+- `fast reduce` operation for work-group, i.e. `reduce_over_group()`
+
+So, if the reduction operation/type has both `fast atomics` and `fast reduce`, then the reduction on work-group with 1 reduction variable is implemented using does the following: the elements inside the work-group are reduced using `ext::oneapi::reduce` and the final result is atomically added to the final/global reduction variable.
+
+```c++
+    // Call user's lambda function or functor. Reducer.MValue gets initialized there.
+    typename Reduction::reducer_type Reducer;
+    KernelFunc(NDIt, Reducer);
+
+    typename Reduction::binary_operation BOp;
+    // Compute the partial sum for the work-group.
+    // Then the 0-th work-iteam stores it to the final/global reduction variable.
+    Reducer.MValue = ext::oneapi::reduce(NDIt.get_group(), Reducer.MValue, BOp);
+    if (NDIt.get_local_linear_id() == 0)
+      Reducer.atomic_combine(Reduction::getOutPointer(Out));
+```
+
+The most general case is when the `fast atomics` and `fast reduce` are not available.
+It computes the partial sum using tree-reduction loop and stores the partial sum for the work-group to a global array of partial sums, which later must be also reduced (by using additional(s) kernel(s)). This general case algorithm also requires allocation of a local memory for the tree-reduction loop:
```

Comment:
```suggestion
It computes the partial sum using tree-reduction loop and stores the partial sum for the work-group to a global array of partial sums, which later must be also reduced (by using additional kernel(s)). This general case algorithm also requires allocation of a local memory for the tree-reduction loop:
```

Reply:
Thank you. Fixed as suggested: 55c3d7537ae5a3ac89729ee4fb88cad41873ab5d